rollup/rollup_sync_service/rollup_sync_service.go

```diff
@@ -225,21 +225,36 @@ func (s *RollupSyncService) parseAndUpdateRollupEventLogs(logs []types.Log, endB
 			batchIndex := event.BatchIndex.Uint64()
 			log.Trace("found new FinalizeBatch event", "batch index", batchIndex)
 
-			parentBatchMeta, chunks, err := s.getLocalInfoForBatch(batchIndex)
-			if err != nil {
-				return fmt.Errorf("failed to get local node info, batch index: %v, err: %w", batchIndex, err)
+			lastFinalizedBatchIndex := rawdb.ReadLastFinalizedBatchIndex(s.db)
+			// When failed to get lastFinalizedBatchIndex, set it's value to previous one of current batch index.
+			// It's backward compatible. (works for both before and after darwin)
+			if lastFinalizedBatchIndex == nil {
+				log.Trace("got nil when reading last finalized batch index.")
+				previousBatchIndex := batchIndex - 1
```

Comment:
When we start syncing a new chain, the first 2 events we see will be `CommitBatch(0)`, `FinalizeBatch(0)`: https://github.com/scroll-tech/scroll/blob/develop/contracts/src/L1/rollup/ScrollChain.sol#L262

It seems to me we will underflow at this line.

Reply:
underflow is not a problem.
It also applies to the case when batch index is 0
https://go.dev/play/p/l6gmkedw5S2

Reply:
Ok, you're right. It's a matter of style but I think something like this would be better:

```go
startBatchIndex := batchIndex

if lastFinalizedBatchIndex != nil {
    startBatchIndex = *lastFinalizedBatchIndex + 1
}

for index := startBatchIndex; ...
```

But you can also leave it as it is.

Reply:
After move the index variable above the for loop, I agree with that your style is more concise, wait a minute

Reply:
It's an interesting discussion for this. I also left a related comment: https://github.com/scroll-tech/go-ethereum/pull/865#discussion_r1657111698
but I still think we can add more code to prevent this.

Reply:
I have changed a style, and the comments are also changed.